python/pyproject.toml

```diff
@@ -1,9 +1,10 @@
 [build-system]
-requires = ["maturin==0.14.17"]
+requires = ["maturin==1.8.6"]
```

Comment:
i opted to keep this pinned to a specific version instead of doing a `~=` or `>=`, just because it feels safer?

i did get a `requires the 2024 edition` thing from rustup and had to upgrade, but based on #3795 that seems like it's intended